docs/explanation/build-plans.rst

```diff
@@ -0,0 +1,56 @@
+Build plans
+===========
+
+A build plan is a list of platforms to build. It can be defined in a project's
+metadata file and filtered with command-line arguments.
+
+Craft-application's default build planner can build multiple platforms.
```

Comment:
I believe when referring to the project we should use Craft Application, Craft Parts, etc

Reply:
Generally, I think it might be a good idea to write these documents from the "point of view" of the application. This should make our lives easier when including them in the craft tools

Reply:
I've reworded this so that it can be included in the tools themselves.